plugins/linux-v4l2/v4l2-output.c

```diff
@@ -125,19 +138,57 @@ static bool try_connect(void *data, const char *device)
 
 	vcam->device = open(device, O_RDWR);
 
-	if (vcam->device < 0)
+	if (vcam->device == -1) {
+		blog(LOG_ERROR, "Failed to open device '%s' (%d)", device,
+		     errno);
 		return false;
+	}
+
+	if (ioctl(vcam->device, VIDIOC_QUERYCAP, &capability) == -1) {
+		blog(LOG_WARNING,
+		     "Failed to query device capabilities on '%s' (%d)", device,
+		     errno);
+		goto fail_close_device;
+	}
 
-	if (ioctl(vcam->device, VIDIOC_QUERYCAP, &capability) < 0)
+	blog(LOG_DEBUG,
+	     "Virtual video device: driver='%s', card='%s', capabilities=0x%x, device_caps=0x%x",
+	     capability.driver, capability.card, capability.capabilities,
+	     capability.device_caps);
+
+	/* The v42loopback driver always supports reporting device node capabilities.
+		Here we just make sure to properly follow the API contract which requires
+		that before accessing the content of capability.device_caps, we first
+		check if the driver does in fact support reporting device capabilities */
+	if (!(capability.capabilities & V4L2_CAP_DEVICE_CAPS)) {
+		blog(LOG_ERROR,
+		     "Driver on device '%s' is unable to report on device node capabilities",
+		     device);
 		goto fail_close_device;
+	}
 
+	if (!(capability.device_caps & V4L2_CAP_VIDEO_OUTPUT)) {
+		blog(LOG_ERROR, "Device '%s' does not support video output",
```

Comment:
This check has already been added to PR #11906 - but failing should not be considered an 'error'; OBS tests devices in order to find one with output capabilities. It expects that some of them won't have output.

Also - see comment at line 148 about the debug output that is available with v4l2loopback.